spec/session_ssl_spec.rb

```diff
@@ -258,6 +257,16 @@
     expect(body.request_method).to be == "GET"
   end
 
+  it "should work when ssl_cert is supplied" do
+    @session.insecure = nil
+    @session.ssl_cert = 'spec/certs/keystore.p12'
+    @sessions.ssl_cert_type = "p12"
+    @session.ssl_key_password = "pkcs12"
```

Comment:
If the password is set incorrectly, what is the exception that is going to be raised? Do we need a separate exception type for this?

Reply:
Yeah, will add a spec for that case

Reply:
```
curl -v --cert ./keystore.p12 --pass pkcs13 https://google.com                                                                                                                                           * Rebuilt URL to: https://google.com/
*   Trying 74.125.68.113...
* TCP_NODELAY set
* Connected to google.com (74.125.68.113) port 443 (#0)
* WARNING: SSL: Certificate type not set, assuming PKCS#12 format.
* SSL: Incorrect password for the certificate "./keystore.p12" and its private key.
* Closing connection 0
```

trying to connect with wrong password,  seems like failed at SSL level

Reply:
Yep, would be useful to check for a proper exception in that case on the Patron end

Reply:
sorry, it's kind of busy in those days, i will catch and update this PR asap